apps/judicial-system/backend/src/app/modules/notification/services/caseNotification/caseNotification.service.ts

```diff
@@ -902,48 +942,64 @@ export class CaseNotificationService extends BaseNotificationService {
+        ? {
+            subject: notifications.acceptedWithoutRuling.subject,
+            body: notifications.acceptedWithoutRuling.defenderBody,
+            props: { subject: { courtCaseNumber: theCase.courtCaseNumber } },
+          }
+        : {
+            subject: notifications.signedRuling.subject,
+            body: notifications.signedRuling.defenderBody,
+            props: {
+              subject: {
+                courtCaseNumber: theCase.courtCaseNumber,
+                isModifyingRuling: Boolean(theCase.rulingModifiedHistory),
+              },
+              body: {
+                isModifyingRuling: Boolean(theCase.rulingModifiedHistory),
+              },
+            },
+          }
+
+    return this.sendEmail({
+      subject: isIndictmentCase(theCase.type)
         ? this.formatMessage(notifications.caseCompleted.subject, {
             courtCaseNumber: theCase.courtCaseNumber,
           })
-        : this.formatMessage(notifications.signedRuling.subject, {
-            courtCaseNumber: theCase.courtCaseNumber,
-            isModifyingRuling: Boolean(theCase.rulingModifiedHistory),
+        : this.formatMessage(rulingMessage.subject, {
+            ...rulingMessage.props.subject,
           }),
-      isIndictmentCase(theCase.type)
+      html: isIndictmentCase(theCase.type)
         ? this.formatMessage(notifications.caseCompleted.defenderBody, {
-            courtCaseNumber: theCase.courtCaseNumber,
-            courtName: applyDativeCaseToCourtName(theCase.court?.name || ''),
+            ...sharedHtmlProps,
             caseIndictmentRulingDecision:
               getHumanReadableCaseIndictmentRulingDecision(
                 theCase.indictmentRulingDecision,
               ),
-            defenderHasAccessToRvg: Boolean(defenderNationalId),
-            linkStart: `<a href="${formatDefenderRoute(
-              this.config.clientUrl,
-              theCase.type,
-              theCase.id,
-            )}">`,
-            linkEnd: '</a>',
           })
-        : this.formatMessage(notifications.signedRuling.defenderBody, {
-            isModifyingRuling: Boolean(theCase.rulingModifiedHistory),
-            courtCaseNumber: theCase.courtCaseNumber,
-            courtName: theCase.court?.name?.replace('dómur', 'dómi'),
-            defenderHasAccessToRvg: Boolean(defenderNationalId),
-            linkStart: `<a href="${formatDefenderRoute(
-              this.config.clientUrl,
-              theCase.type,
-              theCase.id,
-            )}">`,
-            linkEnd: '</a>',
+        : this.formatMessage(rulingMessage.body, {
+            ...rulingMessage.props.body,
+            ...sharedHtmlProps,
           }),
-      defenderName ?? '',
-      defenderEmail ?? '',
-      undefined,
-      Boolean(defenderNationalId) === false,
-    )
+      recipientName: defenderName ?? '',
+      recipientEmail: defenderEmail ?? '',
+      attachments: undefined,
```

Comment:
Consider removing undefined properties as they are no longer needed when passing in an object as a parameter.